### PR TITLE
Add further indices to speed up the queries …

### PR DESCRIPTION
**Pull request by auge8472**
… as discussed [in the project forum](http://mylittleforum.net/forum/index.php?id=9958). Additional commits will follow.

### CODE REVIEW

File: install/install.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
...entries (id int(11) NOT NULL auto_increment, pid int(11) NOT NULL default '0', tid int(11) NOT NULL default '0', uniqid varchar(255) NOT NULL default '', time timestamp NOT NULL default CURRENT_TIMESTAMP, last_reply timestamp NULL DEFAULT NULL, edited timestamp NULL DEFAULT NULL, edited_by int(11) default NULL, user_id int(11) default '0', name varchar(255) NOT NULL default '', subject varchar(255) NOT NULL default '',category int(11) NOT NULL default '0', email varchar(255) NOT NULL default '', hp varchar(255) NOT NULL default '', location varchar(255) NOT NULL default '', ip varchar(128) NOT NULL default '', text text NOT NULL, tags varchar(255) NOT NULL default '', show_signature tinyint(4) default '0', email_notification tinyint(4) default '0', marked tinyint(4) default '0', locked tinyint(4) default '0', sticky tinyint(4) default '0', views int(11) default '0', spam tinyint(4) default '0', spam_check_status tinyint(4) default '0', edit_key varchar(255) NOT NULL default '', PRIMARY KEY (id), UNIQUE KEY id (id), KEY tid (tid),KEY category (category), KEY pid (pid), KEY sticky (sticky), KEY user_id (user_id), KEY spam (spam), KEY time (time)) CHARSET=utf8 COLLATE=utf8_general_ci;
```
Comment (derletztekick) [inline install/install.sql]: I would add a key for `last_reply`, too, see http://mylittleforum.net/forum/index.php?id=9985

File: update/update.sql [hunk 2]
```diff
@@ -338,3 +338,11 @@ CREATE TABLE mlf2_read_entries (user_id int(11) UNSIGNED NOT NULL, posting_id in
 CREATE TABLE mlf2_temp_infos (name varchar(50) NOT NULL, value varchar(255) NOT NULL, time timestamp NULL DEFAULT NULL, PRIMARY KEY (name)) ENGINE=InnoDB CHARSET=utf8 COLLATE=utf8_general_ci;
 UPDATE mlf2_settings SET value = '2.4.0' WHERE name = 'version';
 */
+--------------------------------------------------------------------------------
+-- 2.4.2 to 2.4.3
+/*
+ALTER TABLE mlf2_entries ADD INDEX(`spam`);
+ALTER TABLE mlf2_entries ADD INDEX(`user_id`);
+ALTER TABLE mlf2_entries ADD INDEX(`time`);
+*/
```
Comment (derletztekick) [inline update/update.sql]: ALTER TABLE mlf2_entries ADD INDEX(last_reply);

File: update/update_2.3.5-2.4.php [hunk 3]
```diff
@@ -290,6 +290,18 @@
...
+if(empty($update['errors']) && in_array($settings['version'],array('2.3.5', '2.3.6', '2.3.6.1', '2.3.7', '2.3.99.1', '2.3.99.2', '2.3.99.3', '2.4', '2.4.1', '2.4.2'))) {
+	if(!@mysqli_query($connid, "ALTER TABLE `".$db_settings['forum_table']."` ADD INDEX(`spam`);")) {
+		$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
+	}
+	if(!@mysqli_query($connid, "ALTER TABLE `".$db_settings['forum_table']."` ADD INDEX(`user_id`);")) {
+		$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
+	}
+	if(!@mysqli_query($connid, "ALTER TABLE `".$db_settings['forum_table']."` ADD INDEX(`time`);")) {
+		$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
+	}
```
Comment (derletztekick) [inline update/update_2.3.5-2.4.php]: if(!@mysqli_query($connid, "ALTER TABLE `".$db_settings['forum_table']."` ADD INDEX(`last_reply`);")) {
 		$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
}

**Comment by auge8472** [comment]
Will add the additional indices tomorrow.

**Pull request merged**